website/docs/gen-ai/inference/Neuron/vLLM-rayserve.md

```diff
@@ -0,0 +1,565 @@
+
+## Solution Architecture
+
+In this section, we will delve into the architecture of our solution, which combines Llama-3 model, [Ray Serve](https://docs.ray.io/en/latest/serve/index.html) and [Inferentia2](https://aws.amazon.com/ec2/instance-types/inf2/) on Amazon EKS.
+
+![Llama-3-inf2](../img/llama3-inf2-architecture-vllm.png)
+
+## Deploying the Solution
+
+To get started with deploying `Llama-3-8B-instruct` on [Amazon EKS](https://aws.amazon.com/eks/), we will cover the necessary prerequisites and guide you through the deployment process step by step.
+
+This includes setting up the infrastructure using AWS Inferentia instances and deploying the **Ray cluster**.
+
+<CollapsibleContent header={<h2><span>Prerequisites</span></h2>}>
+Before we begin, ensure you have all the prerequisites in place to make the deployment process smooth and hassle-free.
+Ensure that you have installed the following tools on your machine.
+
+1. [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+2. [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+### Deploy
+
+Clone the repository:
+
+```bash
+git clone https://github.com/awslabs/data-on-eks.git
+```
+
+Navigate into the following directory and run `install.sh` script:
+
+**Important Note:** Ensure that you update the region in the `variables.tf` file before deploying the blueprint.
+Additionally, confirm that your local region setting matches the specified region to prevent any discrepancies.
+For example, set your `export AWS_DEFAULT_REGION="<REGION>"` to the desired region.
+
+```bash
+cd data-on-eks/ai-ml/trainium-inferentia/
+./install.sh
+```
+
+### Verify the resources
+
+Verify the Amazon EKS Cluster
+
+```bash
+aws eks --region us-west-2 describe-cluster --name trainium-inferentia
+```
+
+```bash
+# Creates k8s config file to authenticate with EKS
+aws eks --region us-west-2 update-kubeconfig --name trainium-inferentia
+
+kubectl get nodes # Output shows the EKS Managed Node group nodes
+```
+Verify the Karpenter autoscaler Nodepools
+
+```bash
+kubectl get nodepools
+```
+
+```text
+NAME              NODECLASS
+default           default
+inferentia-inf2   inferentia-inf2
+```
+
+### Verify Neuron Plugin
+
+Neuron device plugin exposes Neuron cores & devices to kubernetes as a resource. Verify the status of the plugin installed by the blueprint. 
+
+```bash
+kubectl get ds neuron-device-plugin-daemonset --namespace kube-system
+```
+```bash
+NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
+neuron-device-plugin-daemonset   2         2         2       2            2           <none>          2d2h
```

Comment:
Please change the daemonset name from `neuron-device-plugin-daemonset` to `neuron-device-plugin`.

Reply:
Sure will work on it.